packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -1164,3 +1164,412 @@ describe('paybackLoan before FortCanningHeight', () => {
+    await testing.generate(1)
+  }
+
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setupForDUSDLoan()
+    const dusdInfo = await testing.rpc.token.getToken('DUSD')
+    dusdId = Object.keys(dusdInfo)[0]
+    key = `v0/token/${dusdId}/payback_dfi`
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should be able to payback DUSD loan using DFI after enabled in setGov and repay correctly after penalty rate has changed, TSLA loan unaffected', async () => {
+    await testing.rpc.masternode.setGov({ [attributeKey]: { [key]: 'true' } })
+    await testing.generate(1)
+
+    // take loan for tsla to ensure it is unaffected
+    await testing.rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: `${tslaLoanAmount}@TSLA`
+    })
+    await testing.generate(1)
+    const tslaTakeLoanBlockHeight = await testing.rpc.blockchain.getBlockCount()
+
+    const blockHeightBefore = await testing.rpc.blockchain.getBlockCount()
+    const dusdInterestPerBlock = new BigNumber(netInterest * dusdLoanAmount / (365 * blocksPerDay))
+    const dusdInterestAmountBefore = dusdInterestPerBlock.multipliedBy(new BigNumber(blockHeightBefore - dusdTakeLoanBlockHeight + 1))
+    const dusdLoanAmountBefore = new BigNumber(dusdLoanAmount).plus(dusdInterestAmountBefore.decimalPlaces(8, BigNumber.ROUND_CEIL))
+
+    const tslaInterestPerBlock = new BigNumber(netInterest * tslaLoanAmount / (365 * blocksPerDay))
+    const tslaInterestAmountBefore = tslaInterestPerBlock.multipliedBy(new BigNumber(blockHeightBefore - tslaTakeLoanBlockHeight + 1))
+    const tslaLoanAmountBefore = new BigNumber(tslaLoanAmount).plus(tslaInterestAmountBefore.decimalPlaces(8, BigNumber.ROUND_CEIL))
+
+    const vaultBefore = await testing.rpc.loan.getVault(vaultId) as VaultActive
+    expect(vaultBefore.loanAmounts).toStrictEqual([`${dusdLoanAmountBefore.toFixed(8)}@DUSD`, `${tslaLoanAmountBefore.toFixed(8)}@TSLA`])
+
+    const dfiPaybackAmount = 100
+    const burnInfoBefore = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoBefore.paybackburn).toStrictEqual(new BigNumber(0))
+    expect(burnInfoBefore.dfipaybackfee).toStrictEqual(new BigNumber(0))
+    expect(burnInfoBefore.dfipaybacktokens).toStrictEqual([])
+
+    let paybackLoanBlockHeight = await testing.rpc.blockchain.getBlockCount()
+    await testing.rpc.loan.paybackLoan({
+      vaultId: vaultId,
+      amounts: `${dfiPaybackAmount}@DFI`,
+      from: vaultOwnerAddress
+    })
+
+    await testing.generate(1)
+
+    // price of dfi to dusd depends on the oracle, in this case 1 DFI = 1 DUSD
+    // the default penalty rate is 1%
+    const defaultPenaltyRate = 0.01
+    let currentBlockHeight = await testing.rpc.blockchain.getBlockCount()
+    const dfiUSDPrice = new BigNumber(1).multipliedBy(1 - defaultPenaltyRate) // (DFIUSD oracle price * (1 - penalty rate is %))
```

Comment:
I think It would be more clear, if this "dfiUSDPrice" variable was changed to something like "dusdPricePerDfi" and DFIUSD should be changed to DefiDollar (DUSD), right ?

Reply:
its kinda tricky. its not exactly dusdPricePerDfi as it has the penalty rate included in it. and also it is not dusd either. oracle is for usd. `effectiveUsdPricePerDfi` maybe? i got no qualm in changing the name but i often find something like dfi in usd price clearer. maybe `effectiveDfiPrice` ? we would imply that the price is in usd